### PR TITLE
Add imports linting

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR add a pre-commit hook linting imports. I start thinking how to fix it next time I got failing tests because I import from `PyQt5` instead of `qtpy` (auto import feature of IDE) and push it to repository and wasting calculation time 

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by sofroniewn** [comment]
I like this idea, I havn't look at details yet, but makes sense to me. Curious how others feel

**Comment by goanpeca** [comment]
I like this idea as well :) 

Is this ready to go @Czaki ?

**Comment by tlambert03** [comment]
I'm not opposed to it if others like it (don't really care personally)

**Review by sofroniewn — APPROVED** [review]
LGTM! I think it's ready to merge, will move it to waiting area in case anyone else has any additional comments

File: setup.cfg [hunk 1]
```diff
@@ -136,3 +136,54 @@ exclude_lines =
...
+[importlinter:contract:4]
+name = "Block import from qt to _qt"
+type = layers
+layers=
+    napari.qt
+    napari._qt
+ignore_imports=
+    napari._qt._tests.test_qt_public_imports -> napari.qt
+    napari._qt._tests.test_qt_public_imports -> napari.qt.threading
+    napari.utils.action_manager -> napari.qt
```
Comment (sofroniewn) [inline setup.cfg:189]: @Czaki can you just remove this whole block to close #2772, I don't think it is as important as the others

**Pull request merged**